### PR TITLE
systemd

### PR DESCRIPTION
**Pull request by scopatz**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

In what is surely one of the most ridiculous recipes, here is systemd (which compiles and builds for me locally).  This is needed to get a modern version of libudev on conda-forge, which is needed to get our bluetooth stack up (bluez, pybluez, etc).

### CODE REVIEW

**Comment by scopatz** [comment]
@conda-forge/staged-recipes - this is ready for review!

File: recipes/systemd/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    - elfutils
+    - libgpg-error
+
+test:
+  commands:
+    - test -f "${PREFIX}/lib/libudev.so"
+
+about:
+  home: https://www.freedesktop.org/wiki/Software/systemd/
+  license: LGPL v2.1
```
Comment (beckermr) [inline recipes/systemd/meta.yaml]: probably good to use the SPDX identifier here

File: recipes/systemd/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/systemd/systemd/archive/v{{ version }}.tar.gz
+  sha256: 0611843c2407f8b125b1b7cb93533bdebd4ccf91c99dffa64ec61556a258c7d1
+  patches:
+    # fixes compilation issue for missing socket decl
+    # from https://bugs.freedesktop.org/show_bug.cgi?id=67075
+    - l2tp.h.patch
```
Comment (chrisburr) [inline recipes/systemd/meta.yaml]: Could these be moved to a `patches` folder to make it easier to find the actual recipe?

File: recipes/systemd/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    - socket-proxyd.c.patch
+    - coredump-vacuum.c.patch
+    # avoid clock_adjtime
+    - timesyncd-manager.c.patch
+    # avoid prlimit
+    - nspawn.c.patch
+
+build:
+  number: 0
+  skip: true  # [not linux]
```
Comment (chrisburr) [inline recipes/systemd/meta.yaml]: Using `python >=3.5` isn't going to work for constraining the version. I guess it should be removed in favour of `skip`?
```suggestion
  skip: true  # [py<35 or not linux]
```

Comment (beckermr) [reply recipes/systemd/meta.yaml]: Why @chrisburr? This is how we ship `noarch: python` all the time.

**Comment by jakirkham** [comment]
cc @mingwandroid

File: recipes/systemd/meta.yaml (hunk 3 above)
Comment (beckermr) [reply recipes/systemd/meta.yaml]: Asking out of curiosity more than anything else. I missed a bunch of stuff in conda 101. :p

Comment (chrisburr) [reply recipes/systemd/meta.yaml]: If it's `noarch: xxxx` there will be a single build on Linux that can be installed on any platform. I think in it's current form[1] conda-smithy will generate a Python 2.7, Python 3.6 and Python 3.7 build as there are no `noarch` specifiers.

[1] Assuming I haven't missed any config files that override things

Comment (beckermr) [reply recipes/systemd/meta.yaml]: Ah interesting. I thought conda smithy was smart enough to not make CI jobs for those and then fix the python version in the conda build config locally

File: recipes/systemd/meta.yaml (hunk 1 above)
Comment (scopatz) [reply recipes/systemd/meta.yaml]: Which is?

Comment (beckermr) [reply recipes/systemd/meta.yaml]: LGPL-2.1

https://spdx.org/licenses/LGPL-2.1.html

Comment (scopatz) [reply recipes/systemd/meta.yaml]: This would be good as a linter hint :wink:

Comment (beckermr) [reply recipes/systemd/meta.yaml]: Yes it would! I'm backed up on other things right now but maybe later!

**Comment by mingwandroid** [comment]
I worry about this. It's a dangerous path IMHO. Better to use cdt packages from centos7 Your system provides it, and it is PID1.

I'm far from expert in systemd but if there's one library in our ecosystem that is clearly in the realm of the system it is this one!.. Or any other init system. 

I don't believe there aren't other better ways. Also I believe the libudev you build on to of this will not have access to system resources! Happy to be wrong though..

File: recipes/systemd/meta.yaml (hunk 1 above)
Comment (CJ-Wright) [reply recipes/systemd/meta.yaml]: Also could you add a fix to the auto-tick bot? This would be great to fix for existing recipes.

**Comment by scopatz** [comment]
>  Better to use cdt packages from centos7 Your system provides it, and it is PID1.

We can't install centos7 CDT packages on conda forge, though, because it is all centos6 based.  I'd be happy to be wrong here

**Comment by isuruf** [comment]
It's not possible at the moment, but a plan is at https://github.com/conda-forge/conda-forge.github.io/issues/900. Other than step3, all the others are easy to do.

**Comment by scopatz** [comment]
So this is a work around for not having CDTs for the current state of the world

**Comment by isuruf** [comment]
Let me clarify, you can have a CDT for cos7 with conda-forge right now. Adding a cos7 docker image will make it usable at build stage and testing.

**Comment by scopatz** [comment]
@isuruf @mingwandroid - Is that what should be done? I have this working now, but if you think that is a better option, I can pursue that instead.  That package wouldn't be installable on our centos6 images, though, right? So all downstream packages would also have to be built on the centos7 image. Is that correct?

**Comment by scopatz** [comment]
Also, I have put in this PR to create the Centos7 image: https://github.com/conda-forge/docker-images/pull/119

**Comment by mingwandroid** [comment]
> I have this working now

That's good, honestly and thanks for doing the work (at the very least it's an option or useful reference), but to what level? What tools did you build? If users run them on an old system will the corrupt their systemd config files by emiting newer config stuff that old systemd doesn't grok?

**Comment by mingwandroid** [comment]
> So all downstream packages would also have to be built on the centos7 image. Is that correct?

No, that is not necessary.

**Comment by scopatz** [comment]
> If users run them on an old system will the corrupt their systemd config files by emiting newer config stuff that old systemd doesn't grok?

Yeah, I am not so concerned about the older systems because I don't mind users having to have newer systems to use these packages. So in some sense this whole effort is about how we express that certain packages require newer kernel versions. 

> No, that is not necessary.

great!

**Comment by isuruf** [comment]
Upto this point, all our packages have been cos6 compatible. How do you tell end-users that you need a cos7 system or newer for a specific package? That's where virtual packages come in.

**Comment by isuruf** [comment]
It should be done in conda like https://github.com/conda/conda/pull/9349 and any package needing a newer GLIBC, should add a `run_constrained` for that virtual package.

**Comment by isuruf** [comment]
Yes, and then any package needing a cos7 has to add it as a `run_constrained` to make them installable in older conda versions.

**Comment by scopatz** [comment]
Well, so should the virtual package be for cos7 or on systemd?

**Comment by isuruf** [comment]
It should be for glibc. See https://github.com/conda-forge/conda-forge.github.io/issues/900#issuecomment-543731539 and https://github.com/conda/conda/blob/e200258259dd47ef131937b17c32a363b64dbec9/conda/base/context.py#L755

**Comment by scopatz** [comment]
PR for virtual package at https://github.com/conda/conda/pull/9358

**Pull request closed without merging**